frontend/app/src/main/java/dk/scheduling/schedulingfrontend/pages/CreateTaskPage.kt

```diff
@@ -72,62 +85,66 @@ fun CreateTaskPage(
             },
             label = { Text("Duration (min.)") },
             singleLine = true,
             keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
-            isError = !numbersOnly,
+            isError = task.duration.initializedAndInvalid(),
         )
 
         // start date & end date
         val dateRangeDialog = remember { mutableStateOf(false) }
-        val datePickerValue = remember { mutableStateOf(DateRange()) }
 
         StandardDateRangePicker(
             closeDialog = { dateRangeDialog.value = false },
-            passingDate = { datePickerValue.value = it },
+            passingDate = {
+                taskSetter(task.copy(dateRange = it.copy()))
+            },
             openDialog = dateRangeDialog.value,
         )
 
-        val dateMsg: String = if (datePickerValue.value.isValidRange()) datePickerValue.value.print() else "no interval selected."
-
-        ClickableCard({ dateRangeDialog.value = true }, "Date interval: $dateMsg")
+        ClickableCard(
+            { dateRangeDialog.value = true },
+            "Date interval: ${task.dateRange.status().msg}",
+            task.dateRange.isInitialized() && !task.dateRange.status().isValid,
+        )
 
         // start time
         val startTimeDialog = remember { mutableStateOf(false) }
-        val startTimeState = rememberTimePickerState()
 
         StandardTimePickerDialog(
             closeDialog = { startTimeDialog.value = false },
-            state = startTimeState,
+            state = task.startTime,
             openDialog = startTimeDialog.value,
         )
-        ClickableCard({ startTimeDialog.value = true }, "Start time: ${formatTime(startTimeState.hour, startTimeState.minute)}")
+        ClickableCard({ startTimeDialog.value = true }, "Start time: ${task.printStartTime()}")
 
         // end time
         val endTimeDialog = remember { mutableStateOf(false) }
-        val endTimeState = rememberTimePickerState()
 
         StandardTimePickerDialog(
             closeDialog = { endTimeDialog.value = false },
-            state = endTimeState,
+            state = task.endTime,
             openDialog = endTimeDialog.value,
         )
-        ClickableCard({ endTimeDialog.value = true }, "End time: ${formatTime(endTimeState.hour, endTimeState.minute)}")
+        ClickableCard({ endTimeDialog.value = true }, "End time: ${task.printEndTime()}")
+
+        if (!task.status().isValid) {
+            Text(
+                text = task.status().msg,
+                color = MaterialTheme.colorScheme.error,
+            )
+        }
 
         // Submit or Cancel
         Button(
             onClick = handleSubmission,
-            enabled = isValidInput(duration, datePickerValue.value, startTimeState, endTimeState),
+            enabled = task.status().isValid,
             modifier = modifier.fillMaxWidth(),
         ) {
             Text(text = "Create task")
```

Comment:
Can you instead add enabled parameter to filled button or outlined button?

Reply:
Done